app/src/main/cpp/ControllerContainer.cpp

```diff
@@ -242,10 +243,43 @@ void ControllerContainer::SetHandJointLocations(const int32_t aControllerIndex,
     }
 #endif
 
+#if defined(SPACES)
+    if (!controller.handMeshMSFT.geometry) {
+        controller.handMeshToggle = Toggle::Create(create);
+        m.root->AddNode(controller.handMeshToggle);
+
+        GeometryDrawablePtr geom = GeometryDrawable::Create(create);
+
+        TransformPtr transform = Transform::Create(create);
+        transform->SetTransform(vrb::Matrix::Identity().TranslateInPlace(kAverageHeight));
+        transform->AddNode(geom);
+
+        RenderBufferPtr renderBuffer = RenderBuffer::Create(create);
+        renderBuffer->DefinePosition(offsetof(HandMeshVertexMSFT, position), 3);
+        renderBuffer->DefineNormal(offsetof(HandMeshVertexMSFT, normal), 3);
+
+        ProgramPtr program = create->GetProgramFactory()->CreateProgram(create, 0);
+        RenderStatePtr state = RenderState::Create(create);
+        state->SetProgram(program);
+        state->SetMaterial(Color(0.5f, 0.5f, 0.5f),
+                           Color(0.5f, 0.5f, 0.5f),
+                           Color(0.5f, 0.5f, 0.5f), 0.75f);
+        state->SetLightsEnabled(true);
+        geom->SetRenderState(state);
+        geom->SetRenderBuffer(renderBuffer);
+
+        controller.handMeshToggle->AddNode(transform);
+        controller.handMeshMSFT.geometry = std::move(geom);
+
+        for (int i = 0; i < 2; i++) {
+            HandMeshBufferPtr buffer = std::make_shared<HandMeshBufferMSFT>();
+            controller.handMeshMSFT.buffers.push_back(buffer);
+        }
+    }
```

Comment:
Again we should move this out of here, this can be easily extracted to some other file.

Reply:
Yes, this should be part of an specialized entity (e.g, a `HandMeshRenderer` class). That's part of the upcoming re-factoring.

Reply:
nice to know